syntax = "proto3";

package flask_grpc;

service FlaskService {
    rpc StringRequest (StringMessage) returns (StringMessage);
    rpc NDArrayRequest (NDArrayMessage) returns (NDArrayMessage);
    rpc ImageRequest (ImageMessage) returns (ImageMessage);
}

// String message
message StringMessage {
    string msg = 1;
}

// NP Array
message NDArrayMessage {
    bytes ndarray = 1;
}

// Image 
message ImageMessage {
    int32 width      = 1;
    int32 height     = 2;
    bytes image_data = 3;
}

message Response {
    string res = 1;
}